src/diffusers/loaders.py

```diff
@@ -67,6 +68,72 @@
 CUSTOM_DIFFUSION_WEIGHT_NAME_SAFE = "pytorch_custom_diffusion_weights.safetensors"
 
 
+class PatchedLoraProjection(nn.Module):
+    def __init__(self, regular_linear_layer, lora_scale=1, network_alpha=None):
+        super().__init__()
+        self.regular_linear_layer = regular_linear_layer
+        self.lora_linear_layer = LoRALinearLayer(
+            self.regular_linear_layer.in_features, self.regular_linear_layer.out_features, network_alpha=network_alpha
+        )
+        self.lora_scale = lora_scale
+
+    def forward(self, input):
+        return self.regular_linear_layer(input) + self.lora_scale * self.lora_linear_layer(input)
+
+
+def text_encoder_attn_modules(text_encoder):
+    attn_modules = []
+
+    if isinstance(text_encoder, CLIPTextModel):
+        for i, layer in enumerate(text_encoder.text_model.encoder.layers):
+            name = f"text_model.encoder.layers.{i}.self_attn"
+            mod = layer.self_attn
+            attn_modules.append((name, mod))
+    elif isinstance(text_encoder, T5EncoderModel):
+        for i, block in enumerate(text_encoder.encoder.block):
+            name = f"encoder.block.{i}.layer.0.SelfAttention"
+            mod = block.layer[0].SelfAttention
+            attn_modules.append((name, mod))
+    else:
+        raise ValueError(f"do not know how to get attention modules for: {text_encoder.__class__.__name__}")
+
+    return attn_modules
+
+
+def text_encoder_lora_state_dict(text_encoder):
+    state_dict = {}
+
+    for name, module in text_encoder_attn_modules(text_encoder):
+        if isinstance(text_encoder, CLIPTextModel):
+            for k, v in module.q_proj.lora_linear_layer.state_dict().items():
+                state_dict[f"{name}.q_proj.lora_linear_layer.{k}"] = v
+
+            for k, v in module.k_proj.lora_linear_layer.state_dict().items():
+                state_dict[f"{name}.k_proj.lora_linear_layer.{k}"] = v
+
+            for k, v in module.v_proj.lora_linear_layer.state_dict().items():
+                state_dict[f"{name}.v_proj.lora_linear_layer.{k}"] = v
+
+            for k, v in module.out_proj.lora_linear_layer.state_dict().items():
+                state_dict[f"{name}.out_proj.lora_linear_layer.{k}"] = v
+        elif isinstance(text_encoder, T5EncoderModel):
+            for k, v in module.q.lora_linear_layer.state_dict().items():
+                state_dict[f"{name}.q.lora_linear_layer.{k}"] = v
+
+            for k, v in module.k.lora_linear_layer.state_dict().items():
+                state_dict[f"{name}.k.lora_linear_layer.{k}"] = v
+
+            for k, v in module.v.lora_linear_layer.state_dict().items():
+                state_dict[f"{name}.v.lora_linear_layer.{k}"] = v
+
+            for k, v in module.o.lora_linear_layer.state_dict().items():
+                state_dict[f"{name}.o.lora_linear_layer.{k}"] = v
+        else:
+            raise ValueError(f"do not know how to get state dict for: {text_encoder.__class__.__name__}")
+
+    return state_dict
```

Comment:
This is used in the training script in lieu of the AttnProcLayers with state dict hacks. It makes it a bit more straightforward to directly map from text encoder to the subset of the state dict that are attention layers

Reply:
I think @patrickvonplaten added `AttnProcLayers()` for that purpose.